openvdb_houdini/houdini/GR_PrimVDBPoints.cc

```diff
@@ -1135,90 +840,85 @@ GR_PrimVDBPoints::inViewFrustum(const UT_Matrix4D& objviewproj)
+    if (!myGeo) return false;
 
     using GridType = openvdb::points::PointDataGrid;
     using TreeType = GridType::TreeType;
     using AttributeSet = openvdb::points::AttributeSet;
 
     const TreeType& tree = grid.tree();
-
-    if (tree.leafCount() == 0)  return false;
+    auto iter = tree.cbeginLeaf();
+    if (!iter) return false;
 
     const int numPoints = myGeo->getNumPoints();
-
-    if (numPoints == 0)         return false;
-
-    TreeType::LeafCIter iter = tree.cbeginLeaf();
+    if (numPoints == 0) return false;
 
     const AttributeSet::Descriptor& descriptor = iter->attributeSet().descriptor();
-
-    const size_t index = descriptor.find(name);
+    const size_t index = descriptor.find(attributeName);
 
     // early exit if attribute does not exist
 
-    if (index == AttributeSet::INVALID_POS)     return false;
+    if (index == AttributeSet::INVALID_POS) return false;
 
     const openvdb::Name type = descriptor.type(index).first;
 
     // fetch vector attribute, if its cache version matches, no upload is required.
 
 #if (UT_VERSION_INT >= 0x0e000000) // 14.0.0 or later
-    RE_VertexArray* bufferGeo = myGeo->findCachedAttrib(r, name.c_str(), RE_GPU_FLOAT16, 3, RE_ARRAY_POINT, true);
+    RE_VertexArray* bufferGeo = myGeo->findCachedAttrib(r, bufferName.c_str(), RE_GPU_FLOAT16, 3, RE_ARRAY_POINT, true);
 #else
-    RE_VertexArray* bufferGeo = myGeo->findCachedAttribOrArray(r, name.c_str(), RE_GPU_FLOAT16, 3, RE_ARRAY_POINT, true);
+    RE_VertexArray* bufferGeo = myGeo->findCachedAttribOrArray(r, bufferName.c_str(), RE_GPU_FLOAT16, 3, RE_ARRAY_POINT, true);
 #endif
 
     if (bufferGeo->getCacheVersion() != version)
     {
         // check if group viewport is in use
 
-        std::string groupName = "";
-        if (openvdb::StringMetadata::ConstPtr s = grid.getMetadata<openvdb::StringMetadata>(openvdb_houdini::META_GROUP_VIEWPORT)) {
-            groupName = s->value();
-        }
+        const openvdb::StringMetadata::ConstPtr s =
+            grid.getMetadata<openvdb::StringMetadata>(openvdb_houdini::META_GROUP_VIEWPORT);
+
+        const std::string groupName = s ? s->value() : "";
         const bool useGroup = !groupName.empty() && descriptor.hasGroup(groupName);
 
         UT_UniquePtr<UT_Vector3H[]> data(new UT_Vector3H[numPoints]);
 
-        std::vector<Name> includeGroups;
-        std::vector<Name> excludeGroups;
-        if (useGroup)   includeGroups.push_back(groupName);
+        std::vector<Name> includeGroups, excludeGroups;
+        if (useGroup) includeGroups.emplace_back(groupName);
 
         MultiGroupFilter filter(includeGroups, excludeGroups, iter->attributeSet());
 
         std::vector<Index64> offsets;
         pointOffsets(offsets, grid.tree(), filter, /*inCoreOnly=*/true);
 
         if (type == "vec3s") {
+            std::vector<Index64> pointOffsets;
+            getPointOffsets(pointOffsets, grid.tree(), includeGroups, excludeGroups, /*inCoreOnly=*/true);
```

Comment:
`getPointOffsets` is a deprecated function, but the vector shadows its replacement, and the vector doesn't seem to get used once it's been populated.  It's not obvious what the intent is here; can you clarify?

Reply:
This is true, good catch. This looks like a possible merge error to me, @Idclip - could the below conversion method not use the offsets built in the call to pointOffsets() above?

Reply:
Yeah must have been a merge error, will fix